script.js

```diff
@@ -1,12 +1,246 @@
+  selectList.value = "Select an episode"; 
 }
 
-window.onload = setup;
+// Level 300
+
+function createEpisodesSelectionList(listOfEpisodes) {
+  let selectList = document.createElement("select");
+  selectList.classList.add("select");
+  let headerElement = document.getElementsByTagName("header")[0];
+  headerElement.appendChild(selectList);
+
+  let option = document.createElement("option");
+  option.classList.add("option");
+  option.textContent = "Select an episode";
+  selectList.appendChild(option);
+
+  for (let episode of listOfEpisodes) {
+    let option = document.createElement("option");
+    option.text = formattedEpisode(episode, false);
+    option.value = listOfEpisodes.indexOf(episode);
+    selectList.appendChild(option);
+  }
+  selectList.addEventListener("change", selectOneEpisode);
+}
+
+function selectOneEpisode() {
+  let listOfEpisodes = allEpisodes;
+  let inputElement = document.getElementsByClassName("input")[0];
+  inputElement.value = "";
+
+  let paragraphElement = document.getElementsByClassName("paragraph")[0];
+  paragraphElement.innerHTML = `Displaying your selection`;
+
+  let selectList = document.getElementsByClassName("select")[0];
+  if (selectList.value === "Select an episode") { 
+    paragraphElement.innerHTML = `Full catalogue`;
+    makePageForEpisodes(listOfEpisodes);
+  } else {
+
+    let selectedEpisode = [];
+    let index = selectList.value;
+    let episodeObject = listOfEpisodes[index];
+    selectedEpisode.push(episodeObject);
+    makePageForEpisodes(selectedEpisode);
+  }
+}
+
+// Level 400
+
+function createShowsSelectionList() {
+  let listOfShows = getAllShows();
+  listOfShows.sort((a, b) => (a.name > b.name ? 1 : -1));
+
+  // When a show is selected, your app should display the episodes
+
+  let selectList = document.createElement("select");
+  selectList.classList.add("select");
+  let headerElement = document.getElementsByTagName("header")[0];
+  headerElement.appendChild(selectList);
+
+  let option = document.createElement("option");
+  option.classList.add("option");
+  option.textContent = "Home - Series Selection";
+  selectList.appendChild(option);
+
+  for (let show of listOfShows) {
+    let option = document.createElement("option");
+    option.text = show.name;
+    option.value = listOfShows.indexOf(show);
+    selectList.appendChild(option);
+  }
+  selectList.addEventListener("change", selectOneShow);
+}
+
+function selectOneShow() {
+  let listOfShows = getAllShows();
+
+  listOfShows.sort((a, b) => (a.name > b.name ? 1 : -1));
+  let inputElement = document.getElementsByClassName("input")[0];
+  inputElement.value = "";
+
+  let paragraphElement = document.getElementsByClassName("paragraph")[0];
+  paragraphElement.innerHTML = `Displaying your selection`;
+
+  let selectList = document.getElementsByClassName("select")[1];
+  if (selectList.value === "Home - Series Selection") {
+    paragraphElement.innerHTML = `Full catalogue`;
+    makePageForEpisodes(listOfShows);
```

Comment:
Your code is now ready to start Level 500 to display all shows once the page is first loaded and also "Home - Series Selection" is selected on dropdown menu.